executor/point_get.go

```diff
@@ -236,13 +248,120 @@ func (e *PointGetExecutor) getAndLock(ctx context.Context, key kv.Key) (val []by
 	if err != nil {
 		return nil, err
 	}
-	val, err = e.get(ctx, key)
-	if err != nil && !kv.ErrNotExist.Equal(err) {
-		return nil, err
+	if !e.ctx.GetSessionVars().IsPessimisticReadConsistency() {
+		val, err = e.get(ctx, key)
+		if err != nil && !kv.ErrNotExist.Equal(err) {
+			return nil, err
+		}
 	}
 	return val, nil
 }
 
+func (e *PointGetExecutor) checkUniqueIndex(key []byte, chk *chunk.Chunk) error {
+	indexValues := make([]types.Datum, 0, 4)
+	for _, idxCol := range e.idxInfo.Columns {
+		colIdx := -1
+		colID := e.tblInfo.Columns[idxCol.Offset].ID
+		var schemaCol *expression.Column
+		for idx, colInSchema := range e.Schema().Columns {
+			if colInSchema.ID == colID {
+				colIdx = idx
+				schemaCol = colInSchema
+				break
+			}
+		}
+		if schemaCol != nil {
+			val := chk.GetRow(0).GetDatum(colIdx, schemaCol.RetType)
+			indexValues = append(indexValues, val)
+		}
+	}
+	conflictErr := kv.ErrWriteConflict.FastGenByArgs(e.startTS, 0, 0, key)
+	if len(indexValues) != len(e.idxVals) {
+		return conflictErr
+	}
+	for i, readlVal := range indexValues {
+		cmpRes, err := readlVal.CompareDatum(e.ctx.GetSessionVars().StmtCtx, &e.idxVals[i])
```

Comment:
If the index value is a prefix or case insensitive, the compare result would not be equal.

Reply:
Prefix index is not allowed to generate point get plans,  `case insensitive` is not supported by now?

Reply:
We could support prefix index point-get in the future.

case insensitive is supported but not enabled by default.
If set `new_collations_enabled_on_first_bootstrap = true` in the config file, it would be enabled.

Reply:
We need to add a test case that would fail when we support prefix index point get in the future.

Reply:
Got